### ISSUE TITLE
Creating an iOS specific build with Xcode's new multiplatform app template doesn't create .ipa

### ISSUE THREAD

**Issue by knlshh** [fastlane/fastlane#20768]
### Discussed in https://github.com/fastlane/fastlane/discussions/20759

<div type='discussions-op-text'>

<sup>Originally posted by **knlshh** October 19, 2022</sup>
I'm creating a new SwiftUI app that uses Xcode's new multiplatform app template that supports building macOS + iOS apps in a single codebase. Please note that this is **not** a Catalyst app. 

The Supported Destinations look like this in Xcode:
<img width="1019" alt="Screen Shot 2022-10-19 at 10 22 15 PM" src="https://user-images.githubusercontent.com/10396271/196796061-a0a5d3f9-c61a-4f4f-9210-00715301a5bc.png">

When I run Fastlane's `gym` step with the platform as `ios`, I'd expect the `.ipa` file to be created, but this isn't the case (archiving succeeds, but the `.ipa` file isn't generated).

If I remove the `Mac` destination from the above list and re-run `gym`, the `.ipa` file is created. I suspect this is because of the `building_for_ios` (https://github.com/fastlane/fastlane/blob/950bb820237d52e15690408083e328f739ac0ef3/gym/lib/gym/module.rb#L28) conditional.

Copied as-is from the above link:
```
def building_for_ios?
  if Gym.project.mac?
    # Can be building for iOS if mac project and catalyst
    return building_mac_catalyst_for_ios?
  else
    # Can be iOS project and build for mac if catalyst
    return false if building_mac_catalyst_for_mac?

    # Can be iOS project if iOS, tvOS, or watchOS
    return Gym.project.ios? || Gym.project.tvos? || Gym.project.watchos?
  end
end
```

While determining if Fastlane is building for `ios`, it checks if the project supports `mac` (`Gym.project.mac`), and if yes, returns `building_mac_catalyst_for_ios`. This isn't necessarily the case since a multiplatform app can be building for iOS **and** support Mac **and** still not be building Mac Catalyst for iOS. 

Wondering if my understanding is incorrect or if this is a bug in Fastlane? 
</div>

**Comment by harrisonf-ledger** [comment]
I am having the exact same issue.

**Comment by knlshh** [comment]
@harrisonf-ledger Did you find a solution for this?

**Comment by harrisonf-ledger** [comment]
@knlshh No, we had to go back to having separate iOS and macOS targets.

**Comment by mbernson** [comment]
I'm also running into this problem with my multiplatform SwiftUI app (supporting iPhone, iPad and Mac).

I was surprised that even when I explicitly specify the platform on the command line (by running my lane like: `fastlane ios screenshots`) an assumption is made somewhere that we're running it for the Mac platform.

**Comment by bfad** [comment]
I am also having this issue. In fact, to get the builds to work at all, I have to add `skip_package_pkg: true` because otherwise it crashes with an error `[!] PKG invalid` during the packaging stage. However, adding that means that packaging stage is skipped altogether, even if I add `skip_package_ipa: false`. If I don't skip the packaging stage, run in verbose mode, and let it generate that error, then I see a log entry: "Exported [MyApp] to: /var/folders/_j/w9jx_nvn55l1qyknr8l9_rk00000gn/T/gym_output20230328-77768-k26nb8", and if I open that folder, I see the IPA has been created, though a PKG, of course, has not.

**Comment by zachwaugh** [comment]
Having this issue as well, anyone have a workaround? I'm curious how everyone is specifying the platform to build for (wasn't sure the correct way to do that), but I'm setting the `sdk` param to `iphoneos`. If that's the correct way, seems like the easiest fix in Fastlane would be to check the current `sdk` in addition to the platform checks it's currently doing when determining if running for iOS. Otherwise, could add a new configuration like `force_platform` to override the automatic platform detection to at least provide an escape hatch.
